### ISSUE TITLE
Create custom transformer for 'GarageFinish'

### ISSUE THREAD

**Issue by SamMartin92** [SamMartin92/milestone-project-heritage-housing-issues#2]

**Comment by SamMartin92** [comment]
Rough idea of class. This will not work however, refer to workaround in DataCleaning notebook:

---

from sklearn.base import BaseEstimator, TransformerMixin

class MyCustomTransformerForGarageFinish(BaseEstimator, TransformerMixin):

  #### here you define the variables you need to parse when you initialize the class
  def __init__(self, variables):
    # we make sure the variable will be a list, even if only one element
    if not isinstance(variables, list): 
      self.variables = [variables]
    else: self.variables = variables

  #### here is where the learning happens. We perform the operation we are interested in,
  #### in this case, calculate the mean
  def fit(self, X, y=None):
    nan_GarageFinish_vals=X[X['GarageFinish'].isna()]
    nan_GarageFinish_index = list(nan_GarageFinish_vals.index.values)
    # we want to keep the mean value in a dictionary
    X[variable] = X[variable].fillna(0) 
    
    return self

  #### here you transform the variables based on what you learned in the .fit()
  #### You can transform into the train set, test set or real-time data
  def transform(self, X,):
    # loop over the variables and .fillna() in a given feature based on the 
    # mean of a given feature
    for feature in self.variables:
      garage_area_value = df.iloc[x,8]
      if garage_area_value == 0:
        X.at[x,variable] = 'None'
      else:
        X.at[x,variable] = 'Unf'
      
    return X

**Comment by SamMartin92** [comment]
This is an unnecessary feature, better performance without it. Closing.
